app/assets/javascripts/application/messages.js

```diff
@@ -0,0 +1,96 @@
+var mumuki = mumuki || {};
+
+(function (mumuki) {
+
+    MessagesService = {
+        $body: function () {
+            return $('body')
+        },
+        $newMessageModal: function () {
+            return $('.new-message-modal');
+        },
+        $newMessageModalComponents: function () {
+            return MessagesService.$newMessageModal().find('.modal-body, .modal-footer');
+        },
+        collapseNewMessageModal: function () {
+            MessagesService.$newMessageModalComponents().toggleClass('hidden');
+        },
+        token: new mumuki.CsrfToken(),
+        setMessages: function (data) {
+            $('.badge-notifications').html(data.messages_count);
+            $('.notifications-box').toggleClass('notifications-box-empty', !data.has_messages);
+            $('.pending-messages-filter').removeClass('pending-messages-filter');
+            $('button.btn-submit').removeClass('disabled');
+            $('.pending-messages-text').remove()
+        },
+        readMessages: function (url) {
+            MessagesService.tokenRequest({
+                url: url,
+                method: 'POST',
+                success: MessagesService.setMessages,
+                xhrFields: {withCredentials: true}
+            })
+        },
+        tokenRequest: function (data) {
+            $.ajax(MessagesService.token.newRequest(data))
+        },
+        getMessages: function () {
+            if ($('.badge-messages').length == 0) {
+                return;
+            }
+            MessagesService.tokenRequest({
+                url: '/messages',
+                type: 'GET'
+            }).done(function (res) {
+                MessagesService.setMessages(res);
+            }).fail(function (_error) {
+                //ignoring error, not important
+            });
+        },
+        setMessagesInterval: function () {
+            setInterval(MessagesService.getMessages, 60000);
```

Comment:
:warning: Magic - but important - number. Extract to a constant which is easier to find and customize.